pkg/controller/acicontainersoperator.go

```diff
@@ -356,19 +365,21 @@ func (t *OperatorHandler) ObjectCreated(obj interface{}) {
 		log.Info("Error in Fetching deploymentsClient...")
 		return
 	}
+
 	t.Deployment, _ = deploymentsClient.Get("aci-containers-controller",metav1.GetOptions{})
 	if t.Deployment == nil {
 		log.Info("aci-containers-controller deployment is nil..returning")
 		return
 	}
 
-	t.Deployment.OwnerReferences = []metav1.OwnerReference{
-					*metav1.NewControllerRef(acicontainersoperator, operators.SchemeGroupVersion.WithKind("AciContainersOperator")),
-				}
-
-	_, err = deploymentsClient.Update(t.Deployment)
-	if err != nil {
-		log.Error(err.Error())
+	if  !t.CheckOwnerReference(t.Deployment.ObjectMeta.OwnerReferences){
+		t.Deployment.OwnerReferences = []metav1.OwnerReference{
+			*metav1.NewControllerRef(acicontainersoperator, operators.SchemeGroupVersion.WithKind("AciContainersOperator")),
+		}
+		_, err = deploymentsClient.Update(t.Deployment)
+		if err != nil {
+			log.Error(err.Error())
```

Comment:
Wondering if we should just proceed (here & in other such cases below), or bail out? If the owner reference does not get successfully set, and if the CR is deleted at exactly at this point, we might end up with orphan resources. Not catastrophic, but perhaps something to consider.